pkg/minishift/util/iso_path.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	minishiftConstants "github.com/minishift/minishift/pkg/minishift/constants"
+	"net/url"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+func GetIsoPath(isoURL string) string {
+	uri, _ := url.Parse(isoURL)
+	uriPathList := strings.Split(uri.Path, "/")
+
+	// This will only work for minishift-centos and b2d iso which is part of minishift repo
+	if len(uriPathList) == 7 {
```

Comment:
This logic is little weird to me to really on it.

Reply:
String splitting is really fragile and obfuscating in this case? What's wrong with regexp for example? BTW, there might be other non-canonical URLs (as part of the CDN) which might be valid as well. We might want to deal with these as well.

Reply:
@hferentschik Can you provide an non-canonical url example so I can even add it in the unit test also?